### ISSUE TITLE
Duplicate in production ( Laravel Vapor )

### ISSUE THREAD

**Issue by cloudstudio** [wire-elements/spotlight#100]
Hello guys! something weird is happening.

Working super super nice in local as allways, when we deploy in production, using laravel vapor + serverless this is what I see:

![Captura de pantalla 2022-12-19 a las 11 14 01](https://user-images.githubusercontent.com/3589377/208402386-693260b7-2d5f-43fd-8da3-99ad04c9a325.png)

As you see is duplicated 3 times, this happend with all commands.

I try creating custom custom provider, or just using default livewire-ui-spotlight.php 

Any ideas?

`<?php

return [

    /*
    |--------------------------------------------------------------------------
    | Shortcuts
    |--------------------------------------------------------------------------
    |
    | Define which shortcuts will activate Spotlight CTRL / CMD + key
    | The default is CTRL/CMD + K or CTRL/CMD + /
    |
     */

    'shortcuts' => [
        'k',
        'slash',
    ],

    /*
    |--------------------------------------------------------------------------
    | Commands
    |--------------------------------------------------------------------------
    |
    | Define which commands you want to make available in Spotlight.
    | Alternatively, you can also register commands in your AppServiceProvider
    | with the Spotlight::registerCommand(Logout::class); method.
    |
     */

    'commands' => [
        App\Spotlight\SearchVehicle::class
    ],

    /*
    |--------------------------------------------------------------------------
    | Include CSS
    |--------------------------------------------------------------------------
    |
    | Spotlight uses TailwindCSS, if you don't use TailwindCSS you will need
    | to set this parameter to true. This includes the modern-normalize css.
    |
     */
    'include_css' => false,

    /*
    |--------------------------------------------------------------------------
    | Include JS
    |--------------------------------------------------------------------------
    |
    | Spotlight will inject the required Javascript in your blade template.
    | If you want to bundle the required Javascript you can set this to false,
    | call 'npm install fuse.js' or 'yarn add fuse.js',
    | then add `require('vendor/livewire-ui/spotlight/resources/js/spotlight');`
    | to your script bundler like webpack.
    |
     */
    'include_js' => true,

];
`

**Comment by PhiloNL** [comment]
Any chance you are using Laravel Octane?

**Comment by cloudstudio** [comment]
Hello @PhiloNL yes! using Laravel Octane, is because of this ?

**Comment by PhiloNL** [comment]
Yes, the following is creating a memory leak: https://github.com/wire-elements/spotlight/blob/master/src/Spotlight.php#L20 

I think changing it to the following will fix it:

```php
    public static function registerCommand(string $command): void
    {
        tap(new $command, function (SpotlightCommand $command) {
            self::$commands[$command->getId()] = $command;
        });
    }
```

I'll see if I can test this later this week.

**Comment by cloudstudio** [comment]
Let me know if I can help testing it, thanks !

**Comment by Krato** [comment]
Any update?

**Comment by polodev** [comment]
can I pass our prefer id value to SpotlightCommand

currrently its look like

```php
public function getId(): string
{
    return md5(static::class) . $this->name;
}
```

redirect to same route while using single command using loop

```php
foreach(Helpers::get_dashboard_pages_list() as $page) {
      \LivewireUI\Spotlight\Spotlight::$commands[] = new \App\Spotlight\RedirectToSpotlightCommand($page);
    }
```

```php
class RedirectToSpotlightCommand extends SpotlightCommand
{
    protected string $name;
    protected string $description;
    protected string $pageUrl;
    protected array $synonyms;
    protected bool $shouldBeShown  = true;

    public function __construct($args)
    {
      // $name, $description, $pageUrl, $synonyms=[]
      $default_args = [
          'name' => 'random-' . rand(1111, 9999),
          'description' => 'random-' . rand(1111, 9999),
          'pageUrl' => '/',
          'synonyms' => [],
      ];
      $args = array_merge( $default_args, $args );
      extract( $args );
      $this->name = $name;
      $this->description = $description;
      $this->pageUrl = $pageUrl;
      $this->synonyms = $synonyms;
    }

    public function execute(Spotlight $spotlight)
    {
        $spotlight->redirectRoute($this->pageUrl);
    }

    public function shouldBeShown(): bool
    {
        return true;
    }
}

```
